feat: Adding Data Zoom enhancements to TimeSeries and MixedTimeSeries type charts.

Dev10-34 opened the pull request:
<!---
Please write the PR title following the conventions at https://www.conventionalcommits.org/en/v1.0.0/
Example:
fix(dashboard): load charts correctly
-->

### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
The current Time Series and Mixed Time Series charts suffer when huge datasets are being used as a large number of data points are available and it causes visual clutter. Superset already solves this issue partially by having the data zoom feature, however, one has to still manually update the zoom at the dashboard level to zoom into their desired and specific portions of dataset.
Added a configuration that allows the user to set the start and end points of the data zoom bar, allowing users to add a configurable scroll bar to their timeseries charts in essence.
Additionally, added the option to clip the Y Axis to max value if needed.
These extra configurations only show up if the Data Zoom feature is enabled.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->
BEFORE:
<img width="1919" height="876" alt="image" src="https://github.com/user-attachments/assets/b1b04046-8f43-4d0e-a6f7-1a581bd1bac9" />
<img width="1916" height="873" alt="image" src="https://github.com/user-attachments/assets/54031cf0-b7d7-423f-8871-892d3812aa8b" />
Unable to load gif within limits, but the data zoom after saving goes back to 0 and 100.
AFTER:
<img width="1919" height="879" alt="Screenshot 2025-07-31 132327" src="https://github.com/user-attachments/assets/42f57b04-753f-48b0-86e4-c86d468136c9" />
<img width="1919" height="866" alt="Screenshot 2025-07-31 132345" src="https://github.com/user-attachments/assets/92b86cb1-f1b1-4f28-ae6e-b79aa2fab7c8" />
<img width="1919" height="875" alt="Screenshot 2025-07-31 132725" src="https://github.com/user-attachments/assets/f0b5a212-80bf-4ea9-b144-44651cbda0d5" />
![Recording 2025-07-31 133313](https://github.com/user-attachments/assets/8115cc26-1daa-4dec-9f64-894bd413c975)


### TESTING INSTRUCTIONS
<!--- Required! What steps can be taken to manually verify the changes? -->

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

Dev10-34 committed:
[feat]: Adding Data Zoom enhancements to TimeSeries and MixedTimeSeries type charts.
Note: Even though Gantt chart has the data zoom feature, the enhancements have not been made for the same.
	modified:   superset-frontend/plugins/plugin-chart-echarts/src/MixedTimeseries/controlPanel.tsx
	modified:   superset-frontend/plugins/plugin-chart-echarts/src/MixedTimeseries/transformProps.ts
	modified:   superset-frontend/plugins/plugin-chart-echarts/src/MixedTimeseries/types.ts
	modified:   superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/Area/controlPanel.tsx
	modified:   superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/Regular/Bar/controlPanel.tsx
	modified:   superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/Regular/Line/controlPanel.tsx
	modified:   superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/Regular/Scatter/controlPanel.tsx
	modified:   superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/Regular/SmoothLine/controlPanel.tsx
	modified:   superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/Step/controlPanel.tsx
	modified:   superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/constants.ts
	modified:   superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/transformProps.ts
	modified:   superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/types.ts
	modified:   superset-frontend/plugins/plugin-chart-echarts/src/controls.tsx

Dev10-34 committed:
fix: Adding default value for zoomableClipYAxis for MixedTimeSeries

Dev10-34 committed:
fix: Added default values properly for some values
	modified:   superset-frontend/plugins/plugin-chart-echarts/src/MixedTimeseries/types.ts

rusackas commented:
Looks like you need to run precommit to get this to pass CI.

Superset uses Git pre-commit hooks courtesy of [pre-commit](https://pre-commit.com/). To install run the following:
```
pip3 install -r requirements/development.txt
pre-commit install
```
A series of checks will now run when you make a git commit.

Alternatively it is possible to run pre-commit by running pre-commit manually:
```
pre-commit run --all-files
```

sadpandajoe reviewed (commented):
Can we add tests to `transformProps.test.ts` for this functionality?

Dev10-34 committed:
fix: Post pre-commit fixes
	modified:   superset-frontend/plugins/plugin-chart-echarts/src/MixedTimeseries/transformProps.ts
	modified:   superset-frontend/plugins/plugin-chart-echarts/src/MixedTimeseries/types.ts
	modified:   superset-frontend/plugins/plugin-chart-echarts/src/controls.tsx
	modified:   superset-frontend/plugins/plugin-chart-echarts/test/MixedTimeseries/transformProps.test.ts

Dev10-34 commented:
Hey @michael-s-molina, @sadpandajoe. Can this get reviewed soon?

sadpandajoe commented:
@Dev10-34 I still don't see any tests for this new feature. There is now the field for formData, but it's set to false. Do you have tests somewhere else for this feature?
